text/0000-nested-publish.md

```diff
@@ -0,0 +1,198 @@
+- Feature Name: `nested_publish`
+- Start Date: 2023-06-30
+- RFC PR: [rust-lang/rfcs#3452](https://github.com/rust-lang/rfcs/pull/3452)
+- Rust Issue: ...
+
+# Summary
+[summary]: #summary
+
+Allow Cargo packages to be bundled within other Cargo packages when they are published (not just in unpublished workspaces).
+
+# Motivation
+[motivation]: #motivation
+
+There are a number of reasons why a Rust developer currently may feel the need to create multiple library crates, and therefore multiple Cargo packages (since one package contains at most one library crate). These multiple libraries could be:
+
+* A trait declaration and a corresponding derive macro (which must be defined in a separate proc-macro library).
+* A library that uses a build script that uses another library or binary (e.g. for precomputation or bindings generation).
+* A logically singular library broken into multiple parts to speed up compilation.
```

Comment:
The whole document talks only about libraries but I assume it would work as well for binaries published to crates.io?

Reply:
Yeah, the language should likely be updated to "'package" to generalize it

Reply:
This is specifically a list of use cases for _multiple libraries_, the thing that you _cannot at all_ currently publish in a single package to crates.io. I've added a secondary list including the use case of binaries sharing a purely private library, and the use case of binaries that are themselves in nested packages will require further design as noted in the "Future possibilities" section.